### DISCUSSION TITLE
How to show outline in Visual Studio using Quarto Extension?

### DISCUSSION THREAD

**Discussion by stuvjordan-uroc** [quarto-dev/quarto-cli#6476] [Q&A]
### Description

Hey folks!

I'm building a quarto html book in Visual Studio using the [Quarto Extension](https://marketplace.visualstudio.com/items?itemName=quarto.quarto).  For some reason, an outline of the current document that I could use to quickly navigate from one section to another is not showing up in Visual Studio's outline window.

Screenshot:

![image](https://github.com/quarto-dev/quarto-cli/assets/78857503/5ebcded3-b49a-4924-a252-786dfeeec380)

(Note the empty outline window in the bottom left corner)

Any ideas?

Thanks!

**Comment by mcanouil** [comment]
I don't know as I did nothing special or did not change any settings for that.
<img width="751" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/93bb30c3-a8cb-41db-9d57-5e7fd566c36c">

Could you try a simple document to rule out some parsing issue?
Are you running on the latest VSCode/VSCodium and Quarto extension?

**Comment by RoyCCWang** [comment]
Hi, I just encountered this problem on `Quarto v1.5.40`, `VS Code v1.89.1`, `Quarto Extension for VS Code v1.113.0`. I'm on Fedora Workstation Release 40. VS Code's Outline view shows nothing when I open some `.qmd` files. I can't seem to reproduce when this happens, but it seems more likely when I clone a long manuscript into a new folder.

My work-around is to insert a blank line anywhere in the document, delete that line, then save the `.qmd` file. This somehow forces VS Code to generate the Outline view. I discuss this from minute 9:45 to 10:30 in [my video](https://youtu.be/yWOMujVz4co).

**Reply by mcanouil** [reply]
Unfortunately, we can't use your video to investigate. We need a reproducible example.

Note that reloading the window or the extension (which is a thing in recent VSCode version) should solve the issue which seems to be only a refresh issue.
The outline is a native markdown VSCode feature by the way.

**Reply by amitbirgit** [reply]
Unfortunately reloading the window/extension does not solve the issue ... however disabling/enabling the extension worked

**Comment by matthewbegun** [comment]
I apologise if this is thread necromancy; however, I had this problem today (2026-01-21, VSCode v1.108.1, extension v1.128.0, Win11) and Google turned up this thread.

I found a temporary fix: use the "Go to symbol" command `Ctrl+Shift+O` (or `@` in the search bar). This seems to populate the outline pane for the rest of the session when used.
